app/src/main/java/infinite_zoomer/model/Stroke.java

```diff
@@ -68,6 +68,12 @@ public Stroke(String serializedData) {
         updateBoundingCircle();
     }
 
+    public Stroke(List<Line> mLines) {
+        this.mLines = mLines;
```

Comment:
We should clone here! See [Collections.copy](https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#copy(java.util.List,%20java.util.List)) and the [ArrayList(Collection) constructor](https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html#ArrayList-java.util.Collection-).

Otherwise, when we modify `lines` (parameter) elsewhere, it will also update the contents of `mLines` within the stroke!

Reply:
I think this is what was causing the `lines.clear()` modifying `Stroke` issue!